### PR TITLE
Result with error functions and ErrorCodes

### PR DESCRIPTION
**Pull request by vsouhrada**
- `Api Break:` `cz.eman.kaal.domain.Result` and subclasses has been moved into `cz.eman.kaal.domain.result.Result`. 
This a new `Result` has a new functionality to remove boilerplate code. Also all classes which are using `Result` has been changed
- Dependencies update:
    - Kotlin 1.3.50
    - Kotlin Coroutines 1.3.2
    - Espresso 3.0.2

### CODE REVIEW

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -6,12 +6,12 @@ import org.gradle.api.JavaVersion
 private object Versions {
     const val supportLib = "1.0.0"
     const val archLifecycle = "2.1.0-alpha03"
```
Comment (smidf) [inline buildSrc/src/main/kotlin/Dependencies.kt]: Use stable version

File: kaal-infrastructure/build.gradle.kts [hunk 2]
```diff
@@ -59,6 +59,10 @@ dependencies {
 
     implementation(Dependencies.Libs.lifecycleExtension)
 
+    // 3rd party
+    implementation(Dependencies.Libs.timberKtx)
+    implementation(Dependencies.Libs.retrofit)
```
Comment (smidf) [inline kaal-infrastructure/build.gradle.kts]: Better use `compileOnly` for limitation transitive dependencies

File: gradle.properties [hunk 3]
```diff
@@ -22,4 +22,4 @@ android.enableJetifier=true
 kotlin.code.style=official
 
 #Core Toolking SDK version
-version=0.3.0
\ No newline at end of file
+version=0.4.20-SNAPSHOT
```
Comment (smidf) [inline gradle.properties]: It is correct version?

File: kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ApiErrorResult.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ *
+ * @author Roman Holomek <roman.holomek@eman.cz>
+ */
+data class ApiErrorResult(
+    val errorCode: ErrorCode,
+    val errorMessage: String? = null,
+    val apiThrowable: Throwable? = null
+) : ErrorResult(errorCode, errorMessage, apiThrowable)
```
Comment (smidf) [inline kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ApiErrorResult.kt]: Better solution with elimination duplicate properties
```
data class ApiErrorResult(
    override val code: ErrorCode,
    override val message: String? = null,
    override val throwable: Throwable? = null
) : ErrorResult(code, message, throwable)
```

File: kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt [hunk 5]
```diff
@@ -43,27 +50,65 @@ sealed class Result<out T : Any> {
      */
     fun getOrNull() = when {
```
Comment (smidf) [inline kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt]: `when` with subject should be used

File: kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/common/Common.kt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import cz.eman.logger.logError
+import retrofit2.Response
+
+/**
+ * @author vsouhrada (vaclav.souhrada@eman.cz)
+ * @since 0.4.0
+ */
+suspend fun <Dto, T> callResult(
+    responseCall: suspend () -> Response<Dto>,
+    errorMessage: String? = null,
```
Comment (smidf) [inline kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/common/Common.kt]: Use lazy evaluation

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 1 above)
Comment (vsouhrada) [reply buildSrc/src/main/kotlin/Dependencies.kt]: done - forgotten commit

File: gradle.properties (hunk 3 above)
Comment (vsouhrada) [reply gradle.properties]: done

File: kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ApiErrorResult.kt (hunk 4 above)
Comment (vsouhrada) [reply kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ApiErrorResult.kt]: Thx for hint,,sorry copy & paste from a real project

**Pull request merged**